iree/turbine/kernel/wave/index_sequence_analysis.py

```diff
@@ -94,3 +100,159 @@ def has_strided_access(node: fx.Node) -> bool:
+    elements_per_thread = getattr(custom, "elements_per_thread", None)
+
+    if isinstance(custom, Reduction):
+        return
+
+    for dim in custom.indexing_dims:
+        index_seq = None
+        for constraint in sorted_constraints:
+            if isinstance(constraint, HardwareConstraint):
+                inputs = None
+                if dim in mma_index:
+                    inputs = (mma_index[dim], elements_per_thread, None)
+                else:
+                    # Assumes vector shapes are associated with workgroup dims.
+                    if dim not in workgroup_constraints:
+                        continue
+                    assert (
+                        dim in constraint.vector_shapes
+                    ), f"Dimension {dim} not found in vector shapes"
+                    if constraint.vector_shapes[dim] == 0:
+                        continue
+                    inputs = (
+                        workgroup_constraints[dim].workgroup_dim,
+                        (
+                            1
+                            if not is_contiguous_dim(
+                                dim,
+                                custom.indexing_dims,
+                                constraint.vector_shapes,
+                            )
+                            else elements_per_thread
+                        ),
+                        compute_stride(
+                            custom.indexing_dims, constraint.vector_shapes, dim
+                        ),
+                    )
+                    if elements_per_thread is None:
+                        # Here we end up with a situation where there will be no thread level
+                        # dependence in the dimensional index.
+                        # TODO: Evaluate if this is a valid case.
+                        continue
+                index_seq = constraint.apply(dim, *inputs, dim in mma_index)
+                if dim in mma_index:
+                    index_seq = specialize_index_sequence(index_seq, mma_slices, custom)
+
+            elif constraint.dim == dim:
+                if index_seq is None:
+                    index_seq = constraint.apply()
+                else:
+                    index_seq.start += constraint.apply().start
+
+        if index_seq is not None:
+            index.update({dim: index_seq})
+        else:
+            index.update({dim: IndexSequence(0, 1, 1)})
+
+    custom.index = index
+
+
+def set_post_expansion_indices(trace: CapturedTrace, constraints: list[Constraint]):
+    """
+    Add offsets to the indices based on the expanded dims.
+    """
+    hw_cons = get_hardware_constraint(constraints)
+
+    def apply_offset(node: fx.Node):
+        custom = get_custom(node)
+        if custom.expanded_dims is None:
+            return False
+        for dim, scale in custom.expanded_dims.items():
+            if dim in custom.index:
+                custom.index[dim].start += scale * hw_cons.vector_shapes[dim]
```

Comment:
What if the dim is in mma_indices? Won't we need 
```
vector_size = hardware_constraints[0].mma_matrix_shapes[
                    mma_indices[constraint.dim]
                ]
```

Reply:
Resolved offline, vector_size and mma_size is now fused inside https://github.com/iree-org/iree-turbine/blob/77b60ef44670bcef372a00fa578ace3c2fa60216/iree/turbine/kernel/wave/utils.py#L225-L231